CodeStarterCamp_Week4/FitnessCenter.swift

```diff
@@ -0,0 +1,133 @@
+            } catch InputError.empty {
+                print("입력되지 않았습니다. 다시 입력해주세요.")
+            } catch InputError.invaildValue {
+                print("없는 번호입니다. 다시 입력해주세요.")
+            } catch InputError.outOfValue {
+                print("값의 갯수가 적거나 큽니다.")
+            } catch {
+                print("알 수 없는 오류입니다. 다시 시도해주세요.")
+            }
+        tryRoutine()
+        }
+    
+    func writeName() throws -> String {
+        print("안녕하세요. \(name)피트니스 센터입니다. 회원님의 이름은 무엇인가요?")
+        guard let inputName = readLine() else { throw InputError.empty }
+        return inputName
+    }
+    
+    func inputBodyCondition() throws -> BodyCondition {
+        print("운동 목표치를 순서대로 알려주세요. [상체근력:  하체근력:  근지구력: ]")
+        print("목표치간 띄어쓰기는 필수입니다. ex) 100 100 100")
+        guard let inputTargetBodyCondition = readLine() else { throw InputError.empty }
+        let targetBodyConditionArray: [Int] = try inputTargetBodyCondition.split(separator: " ").map{
+            guard let number = Int($0) else { throw InputError.invaildValue }
+            return number
+        }
+        
+        if targetBodyConditionArray.count != 3 {
+            throw InputError.outOfValue
+        }
+        
+        let bodyCondition = BodyCondition()
+        
+        return bodyCondition
+    }
+    
+    func tryRoutine() {
+        do {
+            try choiceRoutine()
+        } catch InputError.empty {
+            print("입력되지 않았습니다. 다시 입력해주세요.")
+        } catch InputError.invaildValue {
+            print("없는 번호입니다. 다시 입력해주세요.")
+        } catch InputError.outOfValue {
+            print("값의 갯수가 적거나 큽니다.")
+        } catch {
+            print("알 수 없는 오류입니다. 다시 시도해주세요.")
+        }
+    }
+    
+    func choiceRoutine() throws -> Void {
+        print("몇 번째 루틴으로 운동하시겠어요?")
+        var routineNumber = 1
+        self.listOfRoutine.forEach {
+            print("\(routineNumber). \($0.routineName)")
+            routineNumber += 1
+        }
+        
+        guard let inputRoutine = readLine() else { throw InputError.empty }
+        guard let innerRoutine = Int(inputRoutine) else { throw InputError.invaildValue }
+        
+        switch innerRoutine {
+        case 1..<routineNumber:
+            routineNumber = innerRoutine - 1
+        default: throw InputError.outOfValue
+        }
+        
+        print("몇 세트 반복하시겠어요?")
+        
+        guard let inputSetCount = readLine() else { throw InputError.empty }
+        guard let countNumber = Int(inputSetCount) else { throw InputError.invaildValue }
+        
+        member?.doExercise(for: countNumber, routine: self.listOfRoutine[routineNumber])
+        routineNumber = 1
+        
+        if accomplishCondotion() {
+            print("--------------")
+            print("성공입니다!")
+                member?.bodyCondition.checkYourCondition()
+        } else {
+            print("--------------")
+            print("목표에 도달하지 못 했습니다. 현재 \(member?.name)님의 컨디션은 다음과 같습니다.")
```

Comment:
출력문에 `Optional("이름")` 이렇게 나오는 문제는
출력할 때 옵셔널이 벗겨진 상태가 아니라는 점 때문입니다. 😄 

해결 방법은 간단합니다. 옵셔널 바인딩을 하거나, `nil 병합 연산자(??)`를 사용하면 됩니다.
여기서는 간단하게 nil 병합 연산자 쓰는 게 좋겠네요!

이렇게 하면, 제대로 나오실 겁니다. ㅎㅎㅎ

```suggestion
            print("--------------")
            print("목표에 도달하지 못 했습니다. 현재 '\(member?.name ?? "이름 없음")'님의 컨디션은 다음과 같습니다.")
```